### PR TITLE
Move simple write into raftstore

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What is changed and how it works?

Issue Number: Ref #14575

What's Changed:

```commit-message
Move simple write into raftstore
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/message/simple_write.rs [hunk 1]
```diff
@@ -147,7 +147,7 @@ pub struct DeleteRange<'a> {
 }
 
 #[derive(Debug)]
-pub enum SimpleWrite<'a> {
+pub enum SimpleWriteView<'a> {
```
Comment (overvenus) [inline components/raftstore/src/store/message/simple_write.rs:150]: What does View mean?

File: components/raftstore/src/store/mod.rs [hunk 2]
```diff
@@ -19,9 +19,11 @@ mod async_io;
 mod bootstrap;
 mod compaction_guard;
 mod hibernate_state;
+pub mod message;
 mod peer_storage;
 mod region_snapshot;
 mod replication_mode;
+pub mod response_channel;
```
Comment (BusyJay) [inline components/raftstore/src/store/mod.rs:26]: It should be kept in v2.

File: components/raftstore/src/store/message/simple_write.rs (hunk 1 above)
Comment (BusyJay) [reply components/raftstore/src/store/message/simple_write.rs:150]: The original name looks good to me.

File: components/raftstore/src/store/mod.rs (hunk 2 above)
Comment (BusyJay) [reply components/raftstore/src/store/mod.rs:26]: You can solve it by using a generic.

**Review by overvenus — APPROVED** [review]

File: components/raftstore-v2/src/operation/command/write/mod.rs [hunk 3]
```diff
@@ -90,7 +88,7 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         if !self.serving() {
             return;
         }
-        let bin = SimpleWriteReqEncoder::new(
+        let bin = SimpleWriteReqEncoder::<CmdResChannel>::new(
```
Comment (BusyJay) [inline components/raftstore-v2/src/operation/command/write/mod.rs:91]: You can reexport `pub type SimpleWriteReqEncoder = SimpleWriteReqEncoder<CmdResChannel>;`

File: components/raftstore/src/store/simple_write.rs [hunk 4]
```diff
@@ -263,7 +268,7 @@ impl<'a> SimpleWriteReqDecoder<'a> {
                     buf: &buf[1 + read as usize..],
                 })
             }
-            _ => Err(parse_at(logger, buf, index, term)),
+            _ => Err(parse_data_at(buf, index, &format!("term: {}", term))),
```
Comment (BusyJay) [inline components/raftstore/src/store/simple_write.rs:271]: Should return error and panic outside of the function.

**Review by BusyJay — APPROVED** [review]

**Comment by CalvinNeo** [comment]
These tests just work fine on my mac.

**Review by breezewish — APPROVED** [review]

**Pull request merged**